File: articles/perseverance-of-effort-versus-consistency-of-interests/fortran/grit_facet_score.f90
program grit_facet_score
  implicit none
  real :: perseverance_effort, consistency_interests, grit_total

  perseverance_effort = 0.82
  consistency_interests = 0.71
  grit_total = 0.60 * perseverance_effort + 0.40 * consistency_interests

  print *, "Synthetic grit total score:", grit_total
end program grit_facet_score

File: articles/perseverance-of-effort-versus-consistency-of-interests/rust/grit_facet_score.rs
fn grit_total_score(perseverance_effort: f64, consistency_interests: f64) -> f64 {
    0.60 * perseverance_effort + 0.40 * consistency_interests
}

fn main() {
    println!(
        "Synthetic grit total score: {:.3}",
        grit_total_score(0.82, 0.71)
    );
}
